pkg/parse/parse.go

```diff
@@ -53,16 +53,17 @@ const (
 )
 
 // RepoNamesToNamedReferences parse the raw string to Named reference
-func RepoNamesToNamedReferences(destList []string) ([]reference.Named, error) {
-	var result []reference.Named
+func RepoNamesToNamedReferences(destList []string) ([]types.ImageReference, error) {
+	var result []types.ImageReference
 	for _, dest := range destList {
-		named, err := reference.ParseNormalizedNamed(dest)
+		if t := alltransports.TransportFromImageName(dest); t == nil {
+			// assume no default transport provided in such case append docker
+			dest = "docker://" + dest
```

Comment:
If this remains: There are already over half a dozen copies of a similar fallback code, in many different variants. That should be centralized.

(For a start, there is `util.DefaultTransport`.)